sos/report/plugins/dellrac.py

```diff
@@ -0,0 +1,63 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin, PluginOpt
+import re
+import time
+
+
+class DellRAC(Plugin, IndependentPlugin):
+
+    short_desc = 'Dell Remote Access Controller Administration'
+
+    plugin_name = 'dellrac'
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('srvadmin-idracadm7',)
+
+    option_list = [
+        PluginOpt('debug', default=False, desc='capture support assist data')
+    ]
+
+    racadm = '/opt/dell/srvadmin/bin/idracadm7'
+
+    def setup(self):
+        for subcmd in ['getniccfg', 'getsysinfo']:
+            self.add_cmd_output(
+                '%s %s' % (self.racadm, subcmd),
+                suggest_filename='racadm_%s' % subcmd)
+
+        if self.get_option("debug"):
+            self.do_debug()
+
+    def do_debug(self):
+        self.add_cmd_output(
+            '%s supportassist accepteula',
+            suggest_filename='racadm_supportassist_accepteula')
+
+        logpath = self.get_cmd_output_path()
+        status = self.collect_cmd_output(
+            '%s supportassist collect' % self.racadm,
+            suggest_filename='racadm_supportassist_collect')
+
+        jid = ''
+        if status['status'] == 0:
+            jid = re.search(r'JID_[0-9]*', status['output']).group()
+
+        done_collect = False
+        while not done_collect and jid:
+            time.sleep(10)
+            status = self.collect_cmd_output(
+                '%s jobqueue view -i %s' % (self.racadm, jid),
+                suggest_filename='racadm_jobqueue_view')
+            done_collect = re.search(r'Status=Completed', status['output'])
+
+        self.add_cmd_output(
+            '%s supportassist collect -f %s/sar.zip' % (self.racadm, logpath),
```

Comment:
What exactly is the difference between this and the `supportassist collect` command from above? At first glance it seems the first one is writing output to stdout, which we'll collect anyways, and the second is writing straight to disk?

Reply:
The first command starts collection of required data. This command with `-f` option creates zip of the data collected earlier. It's all writing to disk.

Reply:
So there seems to be some sort of caching occurring in order for subsequent executions to be able to compress earlier generated data. Does the `-f` form not generate the data if it's not already present?

Reply:
No, it doesn’t as far as I know.